### PR TITLE
JDK-8301494: Replace NULL with nullptr in cpu/arm

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory cpu/arm. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301494](https://bugs.openjdk.org/browse/JDK-8301494): Replace NULL with nullptr in cpu/arm

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12322/head:pull/12322` \
`$ git checkout pull/12322`

Update a local copy of the PR: \
`$ git checkout pull/12322` \
`$ git pull https://git.openjdk.org/jdk pull/12322/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12322`

View PR using the GUI difftool: \
`$ git pr show -t 12322`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12322.diff">https://git.openjdk.org/jdk/pull/12322.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/arm/interp_masm_arm.cpp [hunk 1]
```diff
@@ -160,7 +160,7 @@ void InterpreterMacroAssembler::check_and_handle_earlyret() {
     const Register thread_state = R2_tmp;
 
     ldr(thread_state, Address(Rthread, JavaThread::jvmti_thread_state_offset()));
-    cbz(thread_state, L); // if (thread->jvmti_thread_state() == NULL) exit;
+    cbz(thread_state, L); // if (thread->jvmti_thread_state() == null) exit;
```
Comment (jdksjolen) [inline src/hotspot/cpu/arm/interp_masm_arm.cpp:163]: nullptr

File: src/hotspot/cpu/arm/stubGenerator_arm.cpp [hunk 2]
```diff
@@ -2640,20 +2640,20 @@ class StubGenerator: public StubCodeGenerator {
...
-    //  if (src == NULL) return -1;
+    //  if (src == null) return -1;
     __ cbz(src, L_failed);
 
     //  if (src_pos < 0) return -1;
     __ cmp_32(src_pos, 0);
     __ b(L_failed, lt);
 
-    //  if (dst == NULL) return -1;
+    //  if (dst == null) return -1;
```
Comment (jdksjolen) [inline src/hotspot/cpu/arm/stubGenerator_arm.cpp:2656]: nullptr

File: src/hotspot/cpu/arm/templateInterpreterGenerator_arm.cpp [hunk 3]
```diff
@@ -757,7 +757,7 @@ address TemplateInterpreterGenerator::generate_Reference_get_entry(void) {
 
   const int referent_offset = java_lang_ref_Reference::referent_offset();
 
-  // Check if local 0 != NULL
+  // Check if local 0 != null
```
Comment (jdksjolen) [inline src/hotspot/cpu/arm/templateInterpreterGenerator_arm.cpp:760]: nullptr

File: src/hotspot/cpu/arm/templateTable_arm.cpp [hunk 4]
```diff
@@ -4078,8 +4078,8 @@ void TemplateTable::checkcast() {
 
 
 void TemplateTable::instanceof() {
-  // result = 0: obj == NULL or  obj is not an instanceof the specified klass
-  // result = 1: obj != NULL and obj is     an instanceof the specified klass
+  // result = 0: obj == null or  obj is not an instanceof the specified klass
+  // result = 1: obj != null and obj is     an instanceof the specified klass
```
Comment (jdksjolen) [inline src/hotspot/cpu/arm/templateTable_arm.cpp:4083]: nullptr

**Review by jdksjolen — COMMENTED** [review]
Some minor fixes needed.

File: src/hotspot/cpu/arm/interp_masm_arm.cpp [hunk 5]
```diff
@@ -60,7 +60,7 @@ void InterpreterMacroAssembler::call_VM_helper(Register oop_result, address entr
   { Label L;
     ldr(Rtemp, Address(FP, frame::interpreter_frame_last_sp_offset * wordSize));
     cbz(Rtemp, L);
-    stop("InterpreterMacroAssembler::call_VM_helper: last_sp != NULL");
+    stop("InterpreterMacroAssembler::call_VM_helper: last_sp != null");
```
Comment (dholmes-ora) [inline src/hotspot/cpu/arm/interp_masm_arm.cpp:63]: nullptr

**Review by dholmes-ora — APPROVED** [review]
Looks good. Thanks.

**Review by coleenp — APPROVED** [review]

**Review by dholmes-ora — APPROVED** [review]

**Comment by jdksjolen** [comment]
Passes tier1, integrating.

/integrate

**Pull request closed without merging**